InterruptedException thrown by PocketEndpoint

## Steps to reproduce
- First reported when the app had been backgrounded for some time, and was then foregrounded.

### Expected behavior
- Crash should never be thrown.  If the endpoint fails to make a request, it should return a failed response.

### Actual behavior

```
E/AndroidRuntime: FATAL EXCEPTION: RxComputationThreadPool-1
    Process: org.mozilla.tv.firefox.gecko.debug, PID: 13035
    io.reactivex.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | java.lang.InterruptedException
        at io.reactivex.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:367)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onError(ObservableFlatMap.java:293)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:125)
        at io.reactivex.internal.operators.observable.ObservableInterval$IntervalObserver.run(ObservableInterval.java:82)
        at io.reactivex.internal.schedulers.ScheduledDirectPeriodicTask.run(ScheduledDirectPeriodicTask.java:38)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:278)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:273)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
     Caused by: java.lang.InterruptedException
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:75)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:53)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:35)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at org.mozilla.tv.firefox.pocket.PocketEndpoint.request(PocketEndpoint.kt:46)
        at org.mozilla.tv.firefox.utils.PeriodicRequester.requestAsynchronously(PeriodicRequester.kt:90)
        at org.mozilla.tv.firefox.utils.PeriodicRequester.access$requestAsynchronously(PeriodicRequester.kt:38)
        at org.mozilla.tv.firefox.utils.PeriodicRequester$start$1.apply(PeriodicRequester.kt:52)
        at org.mozilla.tv.firefox.utils.PeriodicRequester$start$1.apply(PeriodicRequester.kt:38)
        at io.reactivex.internal.operators.observable.ObservableFlatMap$MergeObserver.onNext(ObservableFlatMap.java:121)
        at io.reactivex.internal.operators.observable.ObservableInterval$IntervalObserver.run(ObservableInterval.java:82) 
        at io.reactivex.internal.schedulers.ScheduledDirectPeriodicTask.run(ScheduledDirectPeriodicTask.java:38) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428) 
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:278) 
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:273) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
        at java.lang.Thread.run(Thread.java:761) 
```

### Device information
* Fire TV device: ?
* Latest affected Firefox version: ?
* Earliest affected Firefox version: ?

--- a/app/src/main/java/org/mozilla/tv/firefox/pocket/PocketEndpoint.kt
+++ b/app/src/main/java/org/mozilla/tv/firefox/pocket/PocketEndpoint.kt
@@ -43,7 +43,14 @@ open class PocketEndpoint(
      * Note that this is a blocking call
      */
     override fun request(): Single<Response<PocketData>> {
-        val videos = runBlocking { getRecommendedVideos() }
+        val videos = try {
+            runBlocking { getRecommendedVideos() }
+        } catch (e: InterruptedException) {
+            // RxJava disposals briefly interrupt their threads, which here will
+            // cause runBlocking to crash. We can treat this as a failed response
+            // without any additional handling.
+            null
+        }
         return when {
             videos == null || videos.isEmpty() -> Single.just(Response.Failure())
             else -> Single.just(Response.Success(videos))

